website/www/site/content/en/documentation/sdks/python-machine-learning.md

```diff
@@ -165,7 +165,53 @@ For detailed instructions explaining how to build and run a pipeline that uses M
 
 ## Beam Java SDK support
 
-RunInference API is available to Beam Java SDK 2.41.0 and later through Apache Beam's [Multi-language Pipelines framework](https://beam.apache.org/documentation/programming-guide/#multi-language-pipelines). Please see [here](https://github.com/apache/beam/blob/master/sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/RunInference.java) for the Java wrapper transform to use and please see [here](https://github.com/apache/beam/blob/master/sdks/java/extensions/python/src/test/java/org/apache/beam/sdk/extensions/python/transforms/RunInferenceTransformTest.java) for some example pipelines.
+The RunInference API is available with the Beam Java SDK versions 2.41.0 and later through Apache Beam's [Multi-language Pipelines framework](https://beam.apache.org/documentation/programming-guide/#multi-language-pipelines). For information about the Java wrapper transform, see [RunInference.java](https://github.com/apache/beam/blob/master/sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/RunInference.java). For example pipelines, see [RunInferenceTransformTest.java](https://github.com/apache/beam/blob/master/sdks/java/extensions/python/src/test/java/org/apache/beam/sdk/extensions/python/transforms/RunInferenceTransformTest.java).
+
+## TensorFlow support
+
+To use TensorFlow with the RunInference API, you need to create a model handler from within `tfx_bsl`, import the required modules, and add the necessary code to your pipline.
+
+First, create a model handler from within `tfx_bsl`. The model handler can be keyed or unkeyed.
+For more information, see [run_inference.py](https://github.com/tensorflow/tfx-bsl/blob/d1fca25e5eeaac9ef0111ec13e7634df836f36f6/tfx_bsl/public/beam/run_inference.py) in the TensorFlow GitHub repository.
+
+```
+tf_handler = CreateModelHandler(inference_spec_type)
+
+# unkeyed
+beam.run_inference(tf_handler)
+
+# keyed
+beam.run_inference(beam.ml.inference.KeyedHandler(tf_handler))
```

Comment:
@ryanthompson591 : this seems broken. I think it will be removed from this PR but we should fix the source:
- It should be KeyedModelHandler 
- I think there is mixing of tfx_bsl.beam and apache_beam (imported as beam). I can imagine this will cause confusion.
- nit: in `inferece_spec_type = model_spec_pb2.InferenceSpecType(saved_model_spec=saved_model_spec)` there is typo in `inference`.

Reply:
@tvalentyn  is right, it should be KeyedModelHandler.

This might be more complete (and I ran this through a colab notebook).

```
    from apache_beam.ml.inference.base import RunInference
    from apache_beam.ml.inference.base import KeyedModelHandler

    tf_handler = CreateModelHandler(inference_spec_type)
    # unkeyed
    RunInference(tf_handler)

    # keyed
    RunInference(KeyedModelHandler(tf_handler))
```

Reply:
I updated the examples, switched to KeyedModelHandler, and fixed the typos (hoping I didn't add new ones).